internal/sync/sync.go

```diff
@@ -383,36 +397,28 @@ func refactorFunc(oldLine, newLine string, filePatterns []string) filepath.WalkF
 }
 
 func excludeFiles(path string, patterns ...string) error {
-	entries, err := os.ReadDir(path)
-	if err != nil {
-		return err
-	}
+	return filepath.WalkDir(path, excludeFunc(path, patterns))
+}
 
-	for _, entry := range entries {
-		joinedPath := filepath.Join(path, entry.Name())
+func excludeFunc(root string, patterns []string) fs.WalkDirFunc {
```

Comment:
This works when removing directories?

Reply:
@trewest I've tried it and it seems to work fine. [Looking at the WalkDir source](https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/path/filepath/path.go;l=441-473) it always calls the function on a path before reading it so this does result in `excludeFunc` being run on a path that does not exist, but since `os.RemoveAll` returns nil on paths that don't exist it'll be fine.

Reply:
Okay, it kept failing for me for some reason when I tried to implement it this way. Thanks